crates/esthri/src/ops/upload.rs

```diff
@@ -10,20 +10,21 @@
  * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
  */
 
-use std::{borrow::Cow, collections::HashMap, path::Path};
+use std::convert::TryInto;
+use std::{borrow::Cow, collections::HashMap, io::SeekFrom, path::Path, sync::Arc};
 
 use bytes::{Bytes, BytesMut};
+use futures::lock::Mutex as stdMutex;
```

Comment:
Generally once I'm "in" the tokio ecosystem I prefer to use tokio versions of things if available, so for a lock tokio::sync::Mutex. Note you don't always need to use a async mutex when working with async code (they explain that in the tokio docs https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html), but in this case you do need to, but we hold the lock "across an await point" (i.e. when we read from the file). But for example, the `PENDING_UPLOADS` lock does not do this, so it can just be a sync mutex.